pkg/kubefnord/unjoin.go

```diff
@@ -0,0 +1,387 @@
+	crClientset, err := util.ClusterRegistryClientset(hostConfig)
+	if err != nil {
+		glog.V(2).Infof("Failed to get cluster registry clientset: %v", err)
+		return err
+	}
+
+	glog.V(2).Info("UnRegistering cluster with the cluster registry.")
+
+	err = unRegisterCluster(crClientset, host, unjoiningClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Could not unregister cluster with the cluster registry: %v", err)
+		return err
+	}
+
+	glog.V(2).Info("UnRegistered cluster with the cluster registry.")
+	return nil
+}
+
+// unRegisterCluster unregisters a cluster with the cluster registry.
+func unRegisterCluster(crClientset *crclient.Clientset, host, unjoiningClusterName string,
+	dryRun bool) error {
+	if dryRun {
+		return nil
+	}
+
+	err := crClientset.ClusterregistryV1alpha1().Clusters().Delete(unjoiningClusterName,
+		&metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deleteFederatedCluster deletes a federated cluster resource that associates
+// the cluster and secret.
+func deleteFederatedCluster(clusterClientset client.Interface, fedClientset *fedclient.Clientset,
+	federationNamespace, unjoiningClusterName string, dryRun bool) (*fedv1a1.FederatedCluster, error) {
+	if dryRun {
+		return nil, nil
+	}
+
+	fedCluster, err := fedClientset.FederationV1alpha1().FederatedClusters().Get(unjoiningClusterName,
+		metav1.GetOptions{})
+	if err != nil {
+		return fedCluster, err
+	}
+
+	err = clusterClientset.CoreV1().Secrets(federationNamespace).Delete(fedCluster.Spec.SecretRef.Name,
+		&metav1.DeleteOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = fedClientset.FederationV1alpha1().FederatedClusters().Delete(unjoiningClusterName,
+		&metav1.DeleteOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// deleteRBACSecret deletes a secret in the unjoining cluster using a service
+// account, and populate that secret into the host cluster to allow it to
+// access the unjoining cluster.
+func deleteRBACSecret(hostClusterClientset, unjoiningClusterClientset client.Interface,
+	namespace, unjoiningClusterName, hostClusterName string, dryRun bool) error {
+
+	glog.V(2).Info("Deleting role binding for service account in unjoining cluster")
+
+	saName := util.ClusterServiceAccountName(unjoiningClusterName, hostClusterName)
+	err := deleteClusterRoleBinding(unjoiningClusterClientset, saName, namespace,
+		unjoiningClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Error deleting role binding for service account in unjoining cluster: %v",
+			err)
+		return err
+	}
+
+	glog.V(2).Info("Deleted role binding for service account in unjoining cluster")
+
+	glog.V(2).Info("Deleting service account in unjoining cluster")
+
+	err = deleteServiceAccount(unjoiningClusterClientset, namespace,
+		unjoiningClusterName, hostClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Error creating service account in joining cluster: %v", err)
```

Comment:
s/creating/deleting  s/joining/unjoining

Reply:
This function is not needed